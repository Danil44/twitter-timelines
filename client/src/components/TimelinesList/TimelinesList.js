import React, { Component } from 'react';
import { Grid, Container } from '@material-ui/core';
import PropTypes from 'prop-types';
import queryString from 'query-string';
import { withRouter } from 'react-router-dom';
import TimelinesItem from '../TimelinesItem/TimelinesItem';

const getQueryParams = history => queryString.parse(history.location.search);
export class TimelinesList extends Component {
  state = {};

  componentDidMount() {
    const { history, getTimelines } = this.props;

    const query = getQueryParams(history);
    const isQueryExists = Object.entries(query).length !== 0;

    if (isQueryExists) {
      getTimelines(query.query);
    }
  }

  render() {
    const { items } = this.props;
    return (
      <Container maxWidth="sm">
        <Grid container spacing={4}>
          {items.map(item => (
            <Grid item xs={12} key={item.id}>
              <TimelinesItem {...item} />
            </Grid>
          ))}
        </Grid>
      </Container>
    );
  }
}

TimelinesList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({}).isRequired).isRequired,
  getTimelines: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
};

export default withRouter(TimelinesList);
