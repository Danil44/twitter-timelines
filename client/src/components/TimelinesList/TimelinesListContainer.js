/* eslint-disable import/no-named-as-default */
import { connect } from 'react-redux';
import TimelinesList from './TimelinesList';
import { getTimelinesFromState } from '../../redux/timelines/timelinesSelectors';
import { getTimelinesStart } from '../../redux/timelines/timelinesActions';

const mapStateToProps = state => ({
  items: getTimelinesFromState(state),
});

const mapDispatchToProps = dispatch => ({
  getTimelines: query => dispatch(getTimelinesStart(query)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TimelinesList);
