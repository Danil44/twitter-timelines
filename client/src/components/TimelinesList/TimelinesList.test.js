import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TimelinesList } from './TimelinesList';

describe('<TimelinesList/> component', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      items: [],
      getTimelines: jest.fn(),
      history: {
        location: {
          search: 'query',
        },
      },
    };
    wrapper = shallow(<TimelinesList {...props} />);
  });

  it('renders without crashing given the required propss', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
