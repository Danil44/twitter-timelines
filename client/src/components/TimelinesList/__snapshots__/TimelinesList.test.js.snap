// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelinesList/> component renders without crashing given the required propss 1`] = `
<WithStyles(ForwardRef(Container))
  maxWidth="sm"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={4}
  />
</WithStyles(ForwardRef(Container))>
`;
